Request: tmpOnTmpfs size option

## Issue description
`boot.tmpOnTmpfs` defaults to the tmpfs defaults: 50% of RAM.
What do you think about making it configurable? tmpfs has no real overhead when not used and just like normal memory, it can be swapped out if memory gets tight.

### Motivation:
I have a hp microserver n40l with 8GB RAM and i can't rebuild the linux kernel with `nixos-rebuild boot -j 1 -I nixpkgs=$HOME/nixpkgs`, as my /tmp fills up even though i have a bit of RAM left.

Any activity on this? I've a build failing, even though boot.tmpOnTmpfs is set to false
```
^[error: builder for '/nix/store/vmvgis89265yki26i34750m3gr698dkd-wezterm-nightly-20210502-nightly.drv' failed with exit code 101;
       last 10 log lines:
       > warning:   379 | }
       > warning:       | ^
       > warning: compilation terminated.
       > warning: harfbuzz/src/hb-ot-layout.cc:1991:1: fatal error: error writing to /build/ccYnkJ0r.s: No space left on device
       > warning:  1991 | }
       > warning:       | ^
       > warning: compilation terminated.
       >
       > LLVM ERROR: IO failure on output stream: No space left on device
       > error: build failed
       For full logs, run 'nix log /nix/store/vmvgis89265yki26i34750m3gr698dkd-wezterm-nightly-20210502-nightly.drv'.
       note: build failure may have been caused by lack of free disk space
error: 1 dependencies of derivation '/nix/store/ybhcyd5slkgfmmb4r3y2hiwrr0zypayq-home-manager-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/38zidvjdgdip4wqxyx1ikm744h82zhvs-home-manager-generation.drv' failed to build
error: 1 dependencies of derivation '/nix/store/9zmhgaigdy9pnfj1281h9zv1qgpiq2vd-user-environment.drv' failed to build
error: 1 dependencies of derivation '/nix/store/rwzb0p96mmm5j5a3qq5c5xvn65jnvjdp-etc.drv' failed to build
error: 1 dependencies of derivation '/nix/store/w6r75i6z5hb1ck3cblx4rr13xhgn5mzh-nixos-system-kartoffel-21.05.20210506.6358647.drv' failed to build
warning: Git tree '/var/src/machine-config' is dirty
building the system configuration...
warning: Git tree '/var/src/machine-config' is dirty
building '/nix/store/vmvgis89265yki26i34750m3gr698dkd-wezterm-nightly-20210502-nightly.drv'...
^C

<pinpox/nixos> on  main [±] via 🌙 took 3m10s
✦ ×  df -h /tmp
Filesystem                                              Size  Used Avail Use% Mounted on
/dev/disk/by-uuid/8dcfb3f0-4dba-4c32-af96-84024706ff76  472G  128G  321G  29% /
```

--- a/nixos/modules/system/boot/tmp.nix
+++ b/nixos/modules/system/boot/tmp.nix
@@ -2,6 +2,9 @@
 
 with lib;
 
+let
+  cfg = config.boot;
+in
 {
 
   ###### interface
@@ -24,18 +27,28 @@ with lib;
       '';
     };
 
+    boot.tmpOnTmpfsSize = mkOption {
+      type = types.oneOf [ types.str types.types.ints.positive ];
+      default = "50%";
+      description = ''
+        Size of tmpfs in percentage.
+        Percentage is defined by systemd.
+      '';
+    };
+
   };
 
   ###### implementation
 
   config = {
 
-    systemd.mounts = mkIf config.boot.tmpOnTmpfs [
+    # When changing remember to update /tmp mount in virtualisation/qemu-vm.nix
+    systemd.mounts = mkIf cfg.tmpOnTmpfs [
       {
         what = "tmpfs";
         where = "/tmp";
         type = "tmpfs";
-        mountConfig.Options = [ "mode=1777" "strictatime" "rw" "nosuid" "nodev" "size=50%" ];
+        mountConfig.Options = [ "mode=1777" "strictatime" "rw" "nosuid" "nodev" "size=${toString cfg.tmpOnTmpfsSize}" ];
       }
     ];
 

--- a/nixos/modules/virtualisation/qemu-vm.nix
+++ b/nixos/modules/virtualisation/qemu-vm.nix
@@ -686,7 +686,7 @@ in
             fsType = "tmpfs";
             neededForBoot = true;
             # Sync with systemd's tmp.mount;
-            options = [ "mode=1777" "strictatime" "nosuid" "nodev" ];
+            options = [ "mode=1777" "strictatime" "nosuid" "nodev" "size=${toString config.boot.tmpOnTmpfsSize}" ];
           };
         "/tmp/xchg" =
           { device = "xchg";

